python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -574,31 +590,96 @@ def resolve_configField(self, _graphene_info: ResolveInfo) -> Optional[GrapheneC
     def resolve_computeKind(self, _graphene_info: ResolveInfo) -> Optional[str]:
         return self._external_asset_node.compute_kind
 
-    def resolve_staleStatus(self, graphene_info: ResolveInfo) -> Any:  # (GrapheneAssetStaleStatus)
-        return self.stale_status_loader.get_status(self._external_asset_node.asset_key)
+    def resolve_staleStatus(
+        self, graphene_info: ResolveInfo, partition: Optional[str] = None
+    ) -> Any:  # (GrapheneAssetStaleStatus)
+        if partition:
+            self._validate_partitions_existence()
+        # The status loader does not support querying for the stale status of a
+        # partitioned asset without specifying a partition, so we return here.
+        return self.stale_status_loader.get_status(self._external_asset_node.asset_key, partition)
+
+    def resolve_staleStatusByPartition(
+        self,
+        graphene_info: ResolveInfo,
+        partitions: Optional[Sequence[str]] = None,
+    ) -> Sequence[Any]:  # (GrapheneAssetStaleStatus)
+        if partitions is None:
+            partitions = self._get_partitions_def().get_partition_keys()
+        else:
+            self._validate_partitions_existence()
+        return [
+            self.stale_status_loader.get_status(self._external_asset_node.asset_key, partition)
+            for partition in partitions
+        ]
+
+    def resolve_staleCauses(
+        self, graphene_info: ResolveInfo, partition: Optional[str] = None
+    ) -> Sequence[GrapheneAssetStaleCause]:
+        if partition:
+            self._validate_partitions_existence()
+        return self._get_staleCauses(partition)
 
-    def resolve_staleCauses(self, graphene_info: ResolveInfo) -> Sequence[GrapheneAssetStaleCause]:
-        causes = self.stale_status_loader.get_stale_root_causes(self._external_asset_node.asset_key)
+    def resolve_staleCausesByPartition(
```

Comment:
This looks awesome!

Is it fair to say this is an `O(n)`  resolver? Just want to make sure the UI doesn't do anything crazy like pass a potentially unbounded partition key list to this.

Reply:
Yeah it should be `O(n)`. This is a new field so it shouldn't have any immediate uncontrollable impact but it should be used with caution. If you fetch it with no arg it will by default try to resolve _all_ partitions.

I figured we are going to need bulk fetch eventually so why not just add it now.